registry/api/protocol/eureka/application.go

```diff
@@ -14,8 +14,52 @@
 
 package eureka
 
+import (
+	"encoding/json"
+	"fmt"
+)
+
 // Application is an array of instances
 type Application struct {
 	Name      string      `json:"name,omitempty"`
 	Instances []*Instance `json:"instance,omitempty"`
 }
+
+// This struct is used ONLY for the specific UnmarshalJSON
```

Comment:
If used only inside the UnmarshalJSON() function, you can define the types inside that function as well.